ledger/src/state/ledger.rs

```diff
@@ -558,8 +558,31 @@ impl<N: Network> LedgerState<N> {
         let amount = Block::<N>::block_reward(block_height);
         let coinbase_transaction = Transaction::<N>::new_coinbase(recipient, amount, rng)?;
 
+        // Filter the transactions to ensure they are new, and append the coinbase transaction.
+        // TODO (howardwu): Improve the performance and design of this.
+        let mut transactions: Vec<Transaction<N>> = transactions
+            .iter()
+            .filter(|transaction| {
```

Comment:
Would a `retain` call be more efficient than a `filter`?

Reply:
the `transactions` is just a slice, though, so it can't be `retain`ed